lib/marshalls/repo.marshall.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+const BaseMarshall = require('./baseMarshall')
+const urlExists = require('url-exists')
+const MARSHALL_NAME = 'repo'
+
+class Marshall extends BaseMarshall {
+  constructor (options) {
+    super(options)
+    this.name = MARSHALL_NAME
+  }
+
+  title () {
+    return 'Identifying package repository...'
+  }
+
+  validate (pkg) {
+    return this.packageRepoUtils.getPackageInfo(pkg.packageName).then(data => {
+      const lastVersionData = data.versions && data.versions[data['dist-tags'].latest]
+      if (lastVersionData && lastVersionData.repository && lastVersionData.repository.url) {
+        const url = lastVersionData.repository.url
+        const cleanUrl = (url.indexOf('//') === -1 ? '' : 'https:') +
+          url.substring(url.indexOf('//'), url.length)
+
+        urlExists(cleanUrl, (err, exists) => {
```

Comment:
this one is a callback and we're mixing callbacks with promises so if we choose to stay with a callback module we need to do it properly (right now that callback is asynchronously executed in the background and outside of the promise context)